### PR TITLE
ignore runtime warning in compute_meta

### LINKED ISSUE #5337: Numpy warnings in compute_meta
I've noticed that the following (just an example) generates a warning (using dask version `2.3.0`):
```python
import dask.array as da
import numpy as np

data0 = da.zeros((3, 10, 10), chunks=(3, 2, 2))
data1 = da.map_blocks(lambda x: np.mean(x, axis=0), data0, dtype=data0.dtype, drop_axis=0)
```
Output:
```
numpy/core/fromnumeric.py:3257: RuntimeWarning: Mean of empty slice
```
This seems related to [this pull request](https://github.com/dask/dask/pull/5103).

Thanks!

<details>
dask==2.3.0
numpy==1.17.0
pkg-resources==0.0.0
toolz==0.10.0
</details>

### PR DESCRIPTION
**Pull request by estebanag**
Fixes #5337 
cc @jrbourbeau 

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/array/utils.py [hunk 1]
```diff
@@ -108,7 +109,9 @@ def meta_from_array(x, ndim=None, dtype=None):
 
 
 def compute_meta(func, _dtype, *args, **kwargs):
-    with np.errstate(all="ignore"):
+    with np.errstate(all="ignore"), warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
```
Comment (mrocklin) [inline dask/array/utils.py]: I'm less sure of my understanding of warnings handling. 

@jcrist, does this look ok to you?  (this review should take only a minute or two I think)

Comment (jcrist) [reply dask/array/utils.py]: This looks correct, but maybe we should silence all warnings instead of just runtime warnings? Not sure.

Comment (jrbourbeau) [reply dask/array/utils.py]: I have a slight preference for just ignoring `RuntimeWarning` for now and broadening to all warnings if the need comes up. But it's probably not a big deal either way.

**Comment by mrocklin** [comment]
OK, merging in.  Thanks @estebanag !

Also, I notice that this is your first code contribution to this repository.  Welcome!

**Comment by estebanag** [comment]
Thanks @mrocklin!

**Pull request merged**